exporter/datadogexporter/metrics_exporter.go

```diff
@@ -126,12 +126,12 @@ func newMetricsExporter(ctx context.Context, params exporter.CreateSettings, cfg
 			cfg.Metrics.TCPAddr.Endpoint,
 			cfg.TimeoutSettings,
 			cfg.LimitedHTTPClientSettings.TLSSetting.InsecureSkipVerify)
-		if err := clientutil.ValidateAPIKey(ctx, cfg.API.Key, params.Logger, apiClient); err != nil && cfg.API.FailOnInvalidKey {
+		if err := clientutil.ValidateAPIKey(ctx, string(cfg.API.Key), params.Logger, apiClient); err != nil && cfg.API.FailOnInvalidKey {
```

Comment:
Will there be a follow up to update clientutil to receive opaque strings?

Reply:
Yes, after we deal with #17373 (right now there are two clients controlled by a feature gate)